packages/web/babel.ts

```diff
@@ -48,7 +48,7 @@ module.exports = function (api: any, options: NxWebBabelPresetOptions = {}) {
               corejs: 3,
               // Do not transform modules to CJS
               modules: false,
-              targets: isModern ? { esmodules: true } : undefined,
+              targets: isModern ? { esmodules: 'intersect' } : undefined,
```

Comment:
Hmm is `{ esmodules: 'intersect' }` a valid `targets`? It doesn't seem like it according to the docs. Please correct me if I am wrong.

`@babel/preset-env` says [`targets` follows the same usage](https://babeljs.io/docs/en/babel-preset-env#targets) as the [top-level `targets`](https://babeljs.io/docs/en/options#targets):

> Type: `string | Array<string> | { [string]: string }`
> Default: `{}`
> ...
> Or an object of minimum environment versions to support:
> ```json
> {
>   "targets": {
>     "chrome": "58",
>     "ie": "11"
>   }
> }
> ```
> Supported environments: `android`, `chrome`, `deno`, `edge`, `electron`, `firefox`, `ie`, `ios`, `node`, `opera`, `rhino`, `safari`, `samsung`.

For [`targets.esmodules`](https://babeljs.io/docs/en/options#targetsesmodules), it says
> Type: `boolean`
> ...
> When the esmodules target is specified, it will intersect with the browsers target and browserslist's targets.

So this PR changed `esmodules` from `true` to `intersect`, which is another truthy value. Did it actually make a difference?

Reply:
Hey! Good question. It's a bit of a hidden option. Had to dig through the source of code of Babel to find it.

I linked and quoted the docs in the commit message, but I see it got squashed. Here is it again:

https://babeljs.io/docs/en/babel-helper-compilation-targets#gettargets
```ts
type InputTargets = { ...Targets, browsers?: Browsers, // When `true`, this completely replaces the `browsers` option. // When `intersect`, this is intersected with the `browsers` // option (giving the higher browsers as the result). esmodules?: boolean | "intersect", };
```

Reply:
(And yes, it makes a difference. I have tested by manually changing the file in `node_modules`. Result in PR description)